From 80206c7cad1e0710d73dde47e3f91031f3b3f89e Mon Sep 17 00:00:00 2001
From: Paul Reioux <reioux@gmail.com>
Date: Tue, 4 Jun 2024 00:01:21 -0700
Subject: [PATCH] arm64: dts: rockchip: change GPU to coarse_demand to enable
 more runtime energy savings

and clean up GPU related dts settings

Signed-off-by: Paul Reioux <reioux@gmail.com>
---
 arch/arm64/boot/dts/rockchip/rk356x.dtsi | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk356x.dtsi b/arch/arm64/boot/dts/rockchip/rk356x.dtsi
index 6ef97c2802c3..415ff216402b 100644
--- a/arch/arm64/boot/dts/rockchip/rk356x.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk356x.dtsi
@@ -626,13 +626,13 @@ gpu: gpu@fde60000 {
 			     <GIC_SPI 41 IRQ_TYPE_LEVEL_HIGH>,
 			     <GIC_SPI 39 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "job", "mmu", "gpu";
-		clocks = <&scmi_clk 1>, <&cru CLK_GPU>;
-		clock-names = "gpu", "bus";
+		clocks = <&cru CLK_GPU>;
+		clock-names = "gpu";
 		resets = <&cru SRST_GPU>;
 		#cooling-cells = <2>;
 		operating-points-v2 = <&gpu_opp_table>;
 		power-domains = <&power RK3568_PD_GPU>;
-		power_policy = "always_on";
+		power_policy = "coarse_demand";
 		status = "disabled";
 		power_model@0 {
 			compatible = "arm,mali-simple-power-model";
-- 
2.34.1

